internal/yggdrasil/yggdrasil.go

```diff
@@ -142,6 +142,7 @@ func (h *Handler) getNamespace(ctx context.Context) string {
 func (h *Handler) GetControlMessageForDevice(ctx context.Context, params yggdrasil.GetControlMessageForDeviceParams) middleware.Responder {
 	deviceID := params.DeviceID
 	if !IsOwnDevice(ctx, deviceID) {
+		h.metrics.IncEdgeDeviceInvalidOwnerCounter()
```

Comment:
this block (lines 144-147) repeats itself 4 times.
but this is one of that cases in which refactoring into a function will not save LOC, but will allow to encapsulate:

Owner validation
Increase metrics counter
Report the same error for this violation
however, I'm not strongly pushing to it. just an idea. going to give my LGTM regardless.

Reply:
Does not repeat entirely, the return object is different:

First instance we return a control message forbidden:
https://github.com/project-flotta/flotta-operator/blob/8dc343aae72560ff35ee060d7620e44d1db6a933/internal/yggdrasil/yggdrasil.go#L144-L146

Here we return a data message forbidden:
https://github.com/project-flotta/flotta-operator/blob/8dc343aae72560ff35ee060d7620e44d1db6a933/internal/yggdrasil/yggdrasil.go#L175-L178

And for the last 2 ones we return a post data message forbidden:

https://github.com/project-flotta/flotta-operator/blob/8dc343aae72560ff35ee060d7620e44d1db6a933/internal/yggdrasil/yggdrasil.go#L218-L221

But I see your point. However, even if they were the same object, I'd still have concerns about readability by encapsulating the metric call in this case, as the other checks in this method wouldn't follow the same trend. Doing it here would create a disparity level in terms of readability/visibility of how the function is read since you don't have to drill down to understand what it does.

This is one case where readability would overrule code replication in my opinion.